commit 3d798167cbc4e00713b621786c71c10ecfd4279b
Author: Todd Whiteman <todd.whiteman@joyent.com>
Date:   2019-06-12T12:52:25-07:00 (4 months ago)
    
    TRITON-1716 Instances migrated with `sdc-migrate` have create_timestamp reset

diff --git a/lib/backends/smartos/tasks/machine_migrate.js b/lib/backends/smartos/tasks/machine_migrate.js
index 8a8f630..28c27cd 100644
--- a/lib/backends/smartos/tasks/machine_migrate.js
+++ b/lib/backends/smartos/tasks/machine_migrate.js
@@ -353,6 +353,41 @@ function restoreZfsQuota(callback) {
 }
 
 
+function setCreateTimestamp(callback) {
+    var self = this;
+
+    var log = self.log;
+    var payload = self.req.params;
+
+    assert.object(payload, 'payload');
+    assert.object(payload.vm, 'payload.vm');
+    assert.string(payload.vm.create_timestamp, 'payload.vm.create_timestamp');
+    assert.uuid(payload.vm_uuid, 'payload.vm_uuid');
+
+    var cmd = '/usr/sbin/zonecfg';
+    var args = [
+        '-z',
+        payload.vm_uuid,
+        util.format('select attr name=create-timestamp; set value=%s; end;',
+            payload.vm.create_timestamp)
+    ];
+
+    log.debug({cmd: cmd, args: args}, 'setCreateTimestamp');
+
+    child_process.execFile(cmd, args, gExecFileDefaults,
+            function _execZonecfgSetTimestampCb(err, stdout, stderr) {
+        if (err) {
+            log.error('zonecfg error:', err, ', stderr:', stderr);
+            self.fatal(new zfsError('zonecfg failure', err, stderr));
+            return;
+        }
+
+        log.debug('setCreateTimestamp:: success');
+        self.finish();
+    });
+}
+
+
 function deleteSnapshot(snapshot, log, callback) {
     assert.string(snapshot, 'snapshot');
     assert.object(log, 'log');
@@ -708,6 +743,8 @@ function start(callback) {
     /* Begin */
     } else if (payload.action === 'get-filesystem-details') {
         getFilesystemDetails.bind(this)(callback);
+    } else if (payload.action === 'set-create-timestamp') {
+        setCreateTimestamp.bind(this)(callback);
 
     /* Sync */
     } else if (payload.action === 'sync' || payload.action === 'receive') {
diff --git a/package.json b/package.json
index 8254166..52664b5 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "cn-agent",
   "description": "Triton Compute Node Agent",
-  "version": "2.12.6",
+  "version": "2.12.7",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
